src/vlogger/vlogger.cpp

```diff
@@ -54,70 +59,63 @@ int g_vlogger_fd = -1;
-static const char *log_names_panic[] = {"panic", "0", NULL};
-static const char *log_names_error[] = {"error", "1", NULL};
-static const char *log_names_warn[] = {"warn", "warning", "2", NULL};
-static const char *log_names_info[] = {"info", "information", "3", NULL};
-static const char *log_names_details[] = {"details", NULL};
-static const char *log_names_debug[] = {"debug", "4", NULL};
-static const char *log_names_fine[] = {"fine", "func", "5", NULL};
-static const char *log_names_finer[] = {"finer",    "func+", "funcall", "func_all",
-                                        "func-all", "6",     NULL};
-static const char *log_names_all[] = {"all", NULL};
+#define COLOR_RED     "\e[0;31m"
+#define COLOR_MAGNETA "\e[2;35m"
+#define COLOR_DEFAULT "\e[0m"
+#define COLOR_GRAY    "\e[2m"
 
 // must be by order because "to_str" relies on that!
-static const level_names levels[] = {
-    {VLOG_NONE, "NONE   ", "\e[0;31m" /*Red*/, (const char **)log_names_none},
-    {VLOG_PANIC, "PANIC  ", "\e[0;31m" /*Red*/, (const char **)log_names_panic},
-    {VLOG_ERROR, "ERROR  ", "\e[0;31m" /*Red*/, (const char **)log_names_error},
-    {VLOG_WARNING, "WARNING", "\e[2;35m" /*Magenta*/, (const char **)log_names_warn},
-    {VLOG_INFO, "INFO   ", "\e[0m" /*Default*/, (const char **)log_names_info},
-    {VLOG_DETAILS, "DETAILS", "\e[0m" /*Default*/, (const char **)log_names_details},
-    {VLOG_DEBUG, "DEBUG  ", "\e[0m" /*Default*/, (const char **)log_names_debug},
-    {VLOG_FINE, "FINE   ", "\e[2m" /*Grey*/, (const char **)log_names_fine},
-    {VLOG_FINER, "FINER  ", "\e[2m" /*Grey*/, (const char **)log_names_finer},
-    {VLOG_ALL, "ALL    ", "\e[2m" /*Grey*/, (const char **)log_names_all},
+static const std::unordered_map<vlog_levels_t, const char *> level_to_color = {
+    {VLOG_NONE, COLOR_RED},        {VLOG_PANIC, COLOR_RED},    {VLOG_ERROR, COLOR_RED},
+    {VLOG_WARNING, COLOR_MAGNETA}, {VLOG_INFO, COLOR_DEFAULT}, {VLOG_DEBUG, COLOR_DEFAULT},
+    {VLOG_FINE, COLOR_GRAY},
+};
+
+static const std::unordered_map<const char *, vlog_levels_t> string_to_level = {
+    {"none", VLOG_NONE},        {"panic", VLOG_PANIC},   {"0", VLOG_PANIC},
+    {"error", VLOG_ERROR},      {"1", VLOG_ERROR},       {"warn", VLOG_WARNING},
+    {"warning", VLOG_WARNING},  {"2", VLOG_WARNING},     {"info", VLOG_INFO},
+    {"information", VLOG_INFO}, {"3", VLOG_INFO},        {"details", VLOG_DETAILS},
+    {"debug", VLOG_DEBUG},      {"4", VLOG_DEBUG},       {"fine", VLOG_FINE},
+    {"func", VLOG_FINE},        {"5", VLOG_FINE},        {"finer", VLOG_FINER},
+    {"func+", VLOG_FINER},      {"funcall", VLOG_FINER}, {"func_all", VLOG_FINER},
+    {"func-all", VLOG_FINER},   {"6", VLOG_FINER},       {"all", VLOG_ALL},
 };
 
 // convert str to vlog_levels_t; upon error - returns the given 'def_value'
 vlog_levels_t from_str(const char *str, vlog_levels_t def_value)
 {
-    size_t num_levels = sizeof(levels) / sizeof(levels[0]);
```

Comment:
can you explain?

Reply:
the old implementation relied on enum values and order.

Now, we conform to doca enum values so the old implementation isn't suffice.

imho - it looks better and much simple now nevertheless.